admin-dev/themes/new-theme/js/app/utils/email-idn.js

```diff
@@ -0,0 +1,41 @@
+/*
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+import punycode from 'punycode';
+
+const $ = global.$;
+
+const init = function initEmailFields() {
+  const $emailFields = $('input[type="email"]');
```

Comment:
To be reusable, `'input[type="email"]'` should be in parameter

Reply:
@PierreRambaud it's a very specific case actually. This would work only for input[type="email"]. Do you see another use case for this?

Reply:
If you have ten forms on your page, you maybe don't want to check them all, they are maybe not using the same validator as you do :thinking: